tracing-error/src/lib.rs

```diff
@@ -117,7 +117,28 @@
     while_true
 )]
 mod backtrace;
+///
+#[cfg(not(feature = "stack-error"))]
+mod heap_error;
 mod layer;
+///
+#[cfg(feature = "stack-error")]
+mod stack_error;
 
 pub use self::backtrace::SpanTrace;
+#[cfg(not(feature = "stack-error"))]
+pub use self::heap_error::TracedError;
 pub use self::layer::ErrorLayer;
+#[cfg(feature = "stack-error")]
+pub use self::stack_error::TracedError;
+
+/// The `tracing-error` prelude.
+///
+/// This brings into scope the `Instrument` and `SpanTraceExt` extension traits that are used to
+/// attach Spantraces to errors and subsequently retrieve them from dyn Errors.
+pub mod prelude {
+    #[cfg(not(feature = "stack-error"))]
+    pub use crate::heap_error::{Instrument as _, IntoTracedError as _, SpanTraceExt as _};
+    #[cfg(feature = "stack-error")]
+    pub use crate::stack_error::{Instrument as _, IntoTracedError as _, SpanTraceExt as _};
```

Comment:
I think we should reexport these traits in `lib.rs` as their actual names, and reexport _that_ in the prelude as `_`. Otherwise, since the module they're defined in is private, there's no public way to name these traits.

Alternatively, the suggestion I had (moving the trait definitions out of the feature flagged module) would solve this, as well as removing the need for feature flagging here.